bin/replace-secrets.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Replaces redacted secret environment variables in an app
+ * directory's `redacted.env` and outputs `.env` with the decrypted values of
+ * the secrets.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const YAML = require('yaml');
+const crypto = require('crypto');
+
+const OUTPUT_ENV_FILE = '.env';
+const REDACTED_ENV_FILE = 'redacted.env';
+const CLOUD_BUILD_FILE = 'cloud_build.yaml';
+
+/**
+ * Reads the Cloud Build config file to identify secrets in the environment.
+ *
+ * @param {string} appDir path to app directory.
+ * @return {Array<string>} list of environment variable names.
+ */
+function identifySecrets(appDir) {
+  const yamlFile = path.join(appDir, CLOUD_BUILD_FILE);
+  try {
+    const yamlContents = fs.readFileSync(yamlFile).toString('utf8');
+    const config = YAML.parse(yamlContents);
+    const secrets = Object.keys(config.secrets[0].secretEnv);
+    console.log(`Identified the following secret env keys: ${secrets}`);
+    return secrets;
+  } catch (e) {
+    // Handle the case where parsing secrets from the cloud build file fails.
+    console.warn('No secret keys were found in cloud_build.yaml.');
+  }
+
+  return [];
+}
+
+/**
+ * Reads the redacted env file and creates a `.env` file including secrets.
+ *
+ * @param {string} appDir path to app directory.
+ */
+function replaceSecrets(appDir) {
+  const envFile = path.join(appDir, REDACTED_ENV_FILE);
+  const envFileContents = fs.readFileSync(envFile).toString('utf8');
+
+  const replacedContents = envFileContents
+    .split('\n')
+    .map(line => {
+      for (const secret of identifySecrets(appDir)) {
+        if (line.startsWith(`${secret}=`)) {
+          const secretVal = process.env[secret];
+          const hash = crypto.createHash('sha256');
+          hash.update(secretVal);
```

Comment:
> ![image](https://user-images.githubusercontent.com/44627152/73218205-8048b500-4127-11ea-8e2e-98ab87830c16.png)

I see that you've changed `sha1` to `sha256`. If that didn't work, you can silence the warning with the following inline comment at the end of this line:

```js
hash.update(secretVal); // lgtm [js/weak-cryptographic-algorithm]
```

Reply:
If you look at the thread above, you'll see I started with `md5`, added that comment, then Justin suggested using SHA256 so the comment wouldn't be necessary and LGTM wouldn't complain :)

Reply:
Aha. Didn't see collapsed comments. Cool!